distribution/sdk/sample-mpegts-streamgenerator/src/main/java/org/codice/alliance/distribution/sdk/video/stream/mpegts/MpegTsUdpClient.java

```diff
@@ -0,0 +1,307 @@
+
+            double delayPerPacket = (double) tsDurationMillis / (double) tsPacketCount;
+
+            long startTime = System.currentTimeMillis();
+
+            int packetsSent = 0;
+
+            try (final FileInputStream fis = new FileInputStream(videoFile)) {
+
+                byte[] buffer = new byte[PACKET_SIZE];
+                int c;
+                while ((c = fis.read(buffer)) != -1) {
+                    bytesSent += c;
+
+                    ChannelFuture cf = ch.writeAndFlush(new DatagramPacket(Unpooled.copiedBuffer(
+                            buffer), new InetSocketAddress(ip, port)));
+
+                    cf.await();
+
+                    packetsSent++;
+
+                    if (packetsSent % 100 == 0) {
+                        Thread.sleep((long) (delayPerPacket * 100));
+                    }
+                    if (packetsSent % 10000 == 0) {
+                        LOGGER.info("Packet sent : {}", packetsSent);
+                    }
+                }
+            }
+
+            long endTime = System.currentTimeMillis();
+
+            LOGGER.info("Time Elapsed: {}", endTime - startTime);
+
+            if (!ch.closeFuture()
+                    .await(100)) {
+                logErrorMessage("Channel time out", false);
+            }
+
+            LOGGER.info("Bytes sent : {} ", bytesSent);
+
+        } catch (InterruptedException | IOException e) {
+            logErrorMessage(String.format("Unable to generate stream : %s", e), false);
+        } finally {
+            // Shut down the event loop to terminate all threads.
+            eventLoopGroup.shutdownGracefully();
+        }
+
+        LOGGER.info("count = " + count.get());
+    }
+
+    private static CommandLine getFFmpegInfoCommand(final String videoFilePath) {
+        final String bundledFFmpegBinaryPath = getBundledFFmpegBinaryPath();
+        File file = new File("target/ffmpeg/" + bundledFFmpegBinaryPath);
+        return new CommandLine(file.getAbsolutePath()).addArgument(SUPPRESS_PRINTING_BANNER_FLAG)
+                .addArgument(INPUT_FILE_FLAG)
+                .addArgument(videoFilePath, HANDLE_QUOTING);
+    }
+
+    private static String getBundledFFmpegBinaryPath() {
+        if (SystemUtils.IS_OS_LINUX) {
+            return "linux/ffmpeg";
+        } else if (SystemUtils.IS_OS_MAC) {
+            return "osx/ffmpeg";
+        } else if (SystemUtils.IS_OS_SOLARIS) {
+            return "solaris/ffmpeg";
+        } else if (SystemUtils.IS_OS_WINDOWS) {
+            return "windows/ffmpeg.exe";
+        } else {
+            throw new IllegalStateException("OS is not Linux, Mac, Solaris, or Windows."
+                    + " No FFmpeg binary is available for this OS, so the plugin will not work.");
+        }
+    }
+
+    private static Duration getVideoDuration(final String videoFilePath) {
+        final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```

Comment:
While `close()` on BAOS is a noop, FindBugs (and I believe most other static analysis tools) aren't quite smart enough to figure that out. Using try-with-resources here is more about quieting those reports. Also, it's just generally good practice to treat every `Closeable` the same way so it becomes habitual for everyone touching the codebase.